homeassistant/components/vicare/climate.py

```diff
@@ -0,0 +1,363 @@
+VICARE_MODE_OFF = "standby"
+
+VICARE_PROGRAM_ACTIVE = "active"
+VICARE_PROGRAM_COMFORT = "comfort"
+VICARE_PROGRAM_ECO = "eco"
+VICARE_PROGRAM_EXTERNAL = "external"
+VICARE_PROGRAM_HOLIDAY = "holiday"
+VICARE_PROGRAM_NORMAL = "normal"
+VICARE_PROGRAM_REDUCED = "reduced"
+VICARE_PROGRAM_STANDBY = "standby"
+
+VICARE_HOLD_MODE_AWAY = "away"
+VICARE_HOLD_MODE_HOME = "home"
+VICARE_HOLD_MODE_OFF = "off"
+
+VICARE_TEMP_WATER_MIN = 10
+VICARE_TEMP_WATER_MAX = 60
+VICARE_TEMP_HEATING_MIN = 3
+VICARE_TEMP_HEATING_MAX = 37
+
+SUPPORT_FLAGS_HEATING = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+SUPPORT_FLAGS_WATER = SUPPORT_TARGET_TEMPERATURE
+
+VICARE_TO_HA_HVAC_HEATING = {
+    VICARE_MODE_DHW: HVAC_MODE_OFF,
+    VICARE_MODE_DHWANDHEATING: HVAC_MODE_AUTO,
+    VICARE_MODE_FORCEDREDUCED: HVAC_MODE_OFF,
+    VICARE_MODE_FORCEDNORMAL: HVAC_MODE_HEAT,
+    VICARE_MODE_OFF: HVAC_MODE_OFF,
+}
+
+HA_TO_VICARE_HVAC_HEATING = {
+    HVAC_MODE_HEAT: VICARE_MODE_FORCEDNORMAL,
+    HVAC_MODE_OFF: VICARE_MODE_FORCEDREDUCED,
+    HVAC_MODE_AUTO: VICARE_MODE_DHWANDHEATING,
+}
+
+VICARE_TO_HA_HVAC_DHW = {
+    VICARE_MODE_DHW: HVAC_MODE_AUTO,
+    VICARE_MODE_DHWANDHEATING: HVAC_MODE_AUTO,
+    VICARE_MODE_FORCEDREDUCED: HVAC_MODE_OFF,
+    VICARE_MODE_FORCEDNORMAL: HVAC_MODE_AUTO,
+    VICARE_MODE_OFF: HVAC_MODE_OFF,
+}
+
+HA_TO_VICARE_HVAC_DHW = {
+    HVAC_MODE_OFF: VICARE_MODE_OFF,
+    HVAC_MODE_AUTO: VICARE_MODE_DHW,
+}
+
+VALUE_UNKNOWN = "unknown"
+
+PYVICARE_ERROR = "error"
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_CIRCUIT, default=-1): int,
+        vol.Optional(CONF_NAME, default="ViCare"): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Create the ViCare climate devices."""
+    from PyViCare import ViCareSession
+
+    if config.get(CONF_CIRCUIT) == -1:
+        vicare_api = ViCareSession(
+            config.get(CONF_USERNAME),
+            config.get(CONF_PASSWORD),
+            "/tmp/vicare_token.save",
+        )
+    else:
+        vicare_api = ViCareSession(
+            config.get(CONF_USERNAME),
+            config.get(CONF_PASSWORD),
+            "/tmp/vicare_token.save",
+            config.get(CONF_CIRCUIT),
+        )
+    add_entities(
+        [
+            ViCareClimate(hass, config.get(CONF_NAME) + " Heating", vicare_api),
+            ViCareWater(hass, config.get(CONF_NAME) + " Water", vicare_api),
+        ]
+    )
+
+
+class ViCareClimate(ClimateDevice):
+    """Representation of the ViCare heating climate device."""
+
+    def __init__(self, hass, name, api):
+        """Initialize the climate device."""
+        self._name = name
+        self._state = None
+        self._api = api
+        self._support_flags = SUPPORT_FLAGS_HEATING
+        self._unit_of_measurement = hass.config.units.temperature_unit
```

Comment:
We should set the unit of the device, not the unit the user uses in home assistant.